peps/pep-0750.rst

```diff
@@ -53,837 +33,1089 @@ deferred pending further experience with f-strings. Work on this PEP was
+may be tempted to use an f-string to embed values into their SQL expression 
+instead of using the ``?`` placeholder and passing the values as a tuple to 
+avoid a `SQL injection attack <https://en.wikipedia.org/wiki/SQL_injection>`__.
 
-    def greet(*args):
-        """Give a static greeting."""
-        return "Hello!"
+More broadly, the inability to transform interpolated values before they are
+combined into a final string limits the utility of f-strings in more complex
+string processing tasks. 
 
-    assert greet"Hi" == "Hello!"  # Use the custom "tag" on the string
+Template strings address these problems by providing 
+developers with access to the string and its interpolated values before they 
+are combined. 
 
-As you can see, ``greet`` is just a callable, in the place that the ``f``
-prefix would go. Let's look at the args:
+For example, imagine we want to generate some HTML. Using template 
+strings, we can define an ``html`` function that allows us to automatically 
+sanitize interpolated values:
 
 .. code-block:: python
 
-    def greet(*args):
-        """Uppercase and add exclamation."""
-        salutation = args[0].upper()
-        return f"{salutation}!"
+    evil = "<script>alert('evil')</script>"
+    template = t"<p>{evil}</p>"
+    assert html(template) == "<p>&lt;script&gt;alert('evil')&lt;/script&gt;</p>"
 
-    greeting = greet"Hello"  # Use the custom "tag" on the string
-    assert greeting == "HELLO!"
-
-The tag function is passed a sequence of arguments. Since our tag string is simply
-``"Hello"``, the ``args`` sequence only contains a string-like value of ``'Hello'``.
-
-With this in place, let's introduce an *interpolation*. That is, a place where
-a value should be inserted:
+Likewise, our hypothetical ``html()`` can make it easy for developers to add 
+attributes to HTML elements using a dictionary:
 
 .. code-block:: python
 
-    def greet(*args):
-        """Handle an interpolation."""
-        # The first arg is the string-like value "Hello " with a space
-        salutation = args[0].strip()
-        # The second arg is an "interpolation"
-        interpolation = args[1]
-        # Interpolations are tuples, the first item is a lambda
-        getvalue = interpolation[0]
-        # It gets called in the scope where it was defined, so
-        # the interpolation returns "World"
-        result = getvalue()
-        recipient = result.upper()
-        return f"{salutation} {recipient}!"
+    attributes = {"src": "shrubbery.jpg", "alt": "looks nice"}
+    template = t"<img {attributes} />"
+    assert html(template) == '<img src="shrubbery.jpg" alt="looks nice" />'
 
-    name = "World"
-    greeting = greet"Hello {name}"
-    assert greeting == "Hello WORLD!"
+Neither of these examples is possible with today's f-strings. By providing a 
+mechanism to intercept and transform interpolated values, template strings
+enable a wide range of string processing use cases.
```

Comment:
This paragraph feels fairly similar to "Template strings address these problems by providing 
developers with access to the string and its interpolated values before they are combined.". Could the paragraphs be combined?

```suggestion
Neither of these examples is possible with f-strings. By providing a 
mechanism to intercept and transform interpolated values, template strings
enable a wide range of string processing use cases.
```

Reply:
I didn't entirely get rid of the duplication but hopefully I've shortened it to the point where it reads better.